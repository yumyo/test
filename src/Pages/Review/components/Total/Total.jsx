import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import BasketCard from '../BasketCard/BasketCard.jsx';

const mapStateToProps = (state) => Object({
    basket: state.basket,
});

const calculateTotal = (basket) =>
    basket.length && basket.reduce((total, item) => {
        return total + (item.price * item.quantity)
    }, 0);

export const TotalComponent = ({ basket }) => (
    <div>
        <h2>Total</h2>
        { `Â£${calculateTotal(basket)}` }
    </div>
);

TotalComponent.propTypes = {
    basket: PropTypes.arrayOf(
        PropTypes.shape({
            price: PropTypes.decimal,
            quantity: PropTypes.number,
        }),
    ).isRequired,
};

export default connect(
    mapStateToProps,
)(TotalComponent);
