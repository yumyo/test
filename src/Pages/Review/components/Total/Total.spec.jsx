import React from 'react';
import { shallow } from 'enzyme';
import { TotalComponent } from './Total.jsx';

describe('<Total />', () => {
    let component;
    const props = {
        basket: [
            {
                price: 10.00,
                quantity: 2,
            },
            {
                price: 12.40,
                quantity: 1,
            }
        ],
    }

    beforeEach(() => {
        component = shallow(<TotalComponent {...props} />);
    });

    it('renders with correct total', () => {
        expect(component).toMatchSnapshot();
    });
});
