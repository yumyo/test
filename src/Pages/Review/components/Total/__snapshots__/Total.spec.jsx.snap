// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> renders with correct total 1`] = `
<div>
  <h2>
    Total
  </h2>
  Â£32.4
</div>
`;
